src/gt4py/next/ffront/fbuiltins.py

```diff
@@ -132,6 +132,31 @@ def builtin_function(fun: Callable[_P, _R]) -> BuiltInFunction[_R, _P]:
     return BuiltInFunction(fun)
 
 
+MaskT = TypeVar("MaskT", bound=Field)
+FieldT = TypeVar("FieldT", bound=Union[Field, gt4py_defs.Scalar, Tuple])
+
+
+class WhereBuiltinFunction(
+    BuiltInFunction[_R, [MaskT, FieldT, FieldT]], Generic[_R, MaskT, FieldT]
+):
+    def __call__(self, mask: MaskT, true_field: FieldT, false_field: FieldT) -> _R:
+        if isinstance(true_field, tuple) or isinstance(false_field, tuple):
+            if not (isinstance(true_field, tuple) and isinstance(false_field, tuple)):
+                raise ValueError(
+                    f"Either both or none can be tuple in {true_field=} and {false_field=}."
+                )
+            if len(true_field) != len(false_field):
+                raise ValueError("Tuple of different size not allowed.")
+            return tuple(where(mask, t, f) for t, f in zip(true_field, false_field))  # type: ignore[return-value] # `tuple` is not `_R`
+        return super().__call__(mask, true_field, false_field)
+
+
+def where_builtin_function(
+    fun: Callable[[MaskT, FieldT, FieldT], _R]
+) -> WhereBuiltinFunction[_R, MaskT, FieldT]:
+    return WhereBuiltinFunction(fun)
```

Comment:
Is this function really needed? Wouldn't work exactly the same using the `WhereBuiltinFunction` class as a decorator (which would call the constructor anyway)?

Reply:
right...